server.js

```diff
@@ -19,4 +20,6 @@ app.post('/api/foods', FoodsController.create);
 app.patch('/api/foods/:id', FoodsController.update);
 app.delete('/api/foods/:id', FoodsController.destroy);
 
+app.get('/api/meals/:category/:year/:month/:day', MealsController.index);
```

Comment:
The reason the date is split into 3 parts is because we need all 3 pieces. 

this route is like 
api/meals/breakfast/2017/05/01

the route you're proposing would be
api/meals/breakfast?date=2017/05/01

the query params would actually make it longer